### PR TITLE
Wow, it's new PR

### PR DESCRIPTION
**Pull request by FylSanya**
1. Add JWT
2. Add atomic transactions
3. Connect Celery, make duration/frequency/price changing, activating/deactivating lots, sending mails by celery tasks.
4. Connect WebSocket to real-time data changing on client side
5. Add History of Lots
6. Start to add tests
7. Some minor fixes

### CODE REVIEW

File: lot/views.py [hunk 1]
```diff
@@ -1,38 +1,88 @@
...
+class LotViewSet(viewsets.ModelViewSet):
+    queryset = Lot.objects.all()
     permission_classes = [IsAuthenticatedOrReadOnly]
-    filter_backends = [filters.OrderingFilter]  # need to fix
-    ordering_fields = ['auction__reserve_price', 'auction__duration']
 
-    def get_serializer(self, *args, **kwargs):
-        return serializers.LotSerializer(*args, **kwargs)
+    def create(self, request):
+        dict_data = request.data.dict()  # convert from QueryDict to dict
```
Comment (greatleroy) [inline lot/views.py:22]: need sync

File: user/models.py [hunk 2]
```diff
@@ -60,11 +60,11 @@ def get_short_name(self):
         return self.username
 
     def _generate_jwt_token(self):
-        dt = datetime.now() + timedelta(days=1)
+        dt = datetime.now() + settings.SIMPLE_JWT['ACCESS_TOKEN_LIFETIME']
```
Comment (greatleroy) [inline user/models.py:63]: :+1:

File: user/serializers.py [hunk 3]
```diff
@@ -1,36 +1,89 @@
...
+    username = serializers.CharField(max_length=255, read_only=True)
+    password = serializers.CharField(max_length=128, write_only=True)
+    token = serializers.CharField(max_length=255, read_only=True)
+
+    def validate(self, data):
+
+        email = data.get('email', None)
+        password = data.get('password', None)
+
+        if email is None:
```
Comment (greatleroy) [inline user/serializers.py:63]: can we replace this validation by `required=True` ?

File: auction/models.py [hunk 4]
```diff
@@ -28,16 +43,80 @@ class TypeClass(models.TextChoices):
...
+        super(Auction, self).save(*args, **kwargs)
 
     class Meta:
         verbose_name = "Auction"
         verbose_name_plural = "Auctions"
+
+
+class HistoryLot(models.Model):
+    auction = models.ForeignKey(Auction, on_delete=models.CASCADE)
+    user_owner = models.ForeignKey(User, on_delete=models.CASCADE, related_name='user_owner')
```
[the remark this answers is not in the document]
Comment (FylSanya) [reply auction/models.py:120]: But that's not a mistake, is it?

**Pull request merged**